src/index.ts

```diff
@@ -245,21 +299,15 @@ export const confirmTransaction = async (
   signature: string,
 ): Promise<string> => {
   const block = await connection.getLatestBlockhash();
-  const result = await connection.confirmTransaction(
+  const rpcResponse = await connection.confirmTransaction(
     {
       signature,
       ...block,
     },
     "confirmed",
   );
 
-  // Note: `confirmTransaction` does not throw an error if the confirmation does not succeed,
-  // but rather a `TransactionError` object. so we handle that here
-  // See https://solana-labs.github.io/solana-web3.js/classes/Connection.html#confirmTransaction.confirmTransaction-1
-  const error = result.value.err;
-  if (error) {
-    throw Error(error.toString());
-  }
```

Comment:
I moved this out to a generic function since we use it in a few places and I don't want to keep cutting and pasting the same explanation.